### DISCUSSION TITLE
About PI30 Q1

### DISCUSSION THREAD

**Discussion by zholdak** [softwarecrash/Solar2MQTT-ESP8266#144] [General]
Hi. I have two inverters. First, Anern-AN-SCI-EVO-4200 returns `00 00 00 000 026 021 026 00 00 000 0030 0000 13` in response of `Q1`. This answer have len of 47 symbols and parsed, in common, correctly, but... The second inverter is EASun-ISolar-SMH-II-2.2K returns `07792 00001 00 00 00 000 034 030 000 01 00 000 0030 0000 0000 00.00 13` in response of `Q1`. This answer have len of 70 symbols and cann't be parsed, because in the code we expect only strings with len of 47 or 105...

What I see. 

Fisrt answer, with len of 47, in my opinion, should be parsed as:
`00 00 01 000 026 021 026 00 00 000 0030 0000 13`:
1. (SCC Flag)
2. (AllowSccOnFlag)
3. (ChargeAverageCurrent)
4. (SCC PWM temperature)
5. Inverter temperature
6. Battery temperature
7. Transformer temperature
8. (GPIO13)
9. Fan lock status
10. (Not used)
11. Fan PWM speed
12. (???)
13. Inverter charge status

Second answer, with len of 70, in my opinion, should be parsed as:
`07792 00001 00 00 00 000 034 030 000 01 00 000 0030 0000 0000 00.00 13`
1. (Time until the end of absorb charging)
2. (Time until the end of float charging)
3. (SCC Flag)
4. (AllowSccOnFlag)
5. (ChargeAverageCurrent)
6. (SCC PWM temperature)
7. Inverter temperature
8. Battery temperature
9. Transformer temperature
10. (GPIO13)
11. Fan lock status
12. (Not used)
13. Fan PWM speed
14. (SCC charge power)
15. (Parallel Warning)
16. (Sync frequency)
17. Inverter charge status

@softwarecrash, what do you think, is any sense to correct that? I am parse, for now, Q1 manually from RAW response and build sensors in that way...

P.S. And.. how does 105-symbols answer looks like?

**Comment by softwarecrash** [comment]
Hey,
The Q1 command is a undocumented debug Thing, to know what is what its Bit dificult.
The min and max length of the command answer is Just a Error catch, in some cases the devices give garbage as answer that doenst fit into this but curios the CRC Match, thats the reason for the limiters, when you have a device that gives correct Data with another length, its No Problem to adjust the range, or make a new Sort list deepens on the answer length.
I dont know a 105 answer, thats Just a value how i think "thats okay" so i cant find any ducumentation for real answer for all the devices.
The pi30 is a Bit wired and i have Not found a nicer way to do that.

**Comment by softwarecrash** [comment]
@zholdak get a actual asnwer from a 105 length Q1 it looks like
`0 00026 00 00 00 000 042 030 040 02 00 000 0030 0000 0000 49.95 10 0 000 000 000 000 00.00 000 000 0 0000`
and comes from a device with PI18
